address/address_test.go

```diff
@@ -544,3 +555,42 @@ func FuzzAddressDecode(f *testing.F) {
 		_ = a.Decode(bytes.NewReader(data))
 	})
 }
+
+// TestAddressUnknownOddType tests that an unknown odd type is allowed in an
+// address and that we can still arrive at the correct leaf hash with it.
+func TestAddressUnknownOddType(t *testing.T) {
+	knownAddr, _, _ := RandAddr(t, &TestNet3Tap, RandProofCourierAddr(t))
+	knownAddrString, err := knownAddr.EncodeAddress()
+	require.NoError(t, err)
+
+	test.RunUnknownOddTypeTest(
+		t, knownAddr.Tap, &asset.ErrUnknownType{},
+		func(buf *bytes.Buffer, addr *Tap) error {
+			return addr.Encode(buf)
+		},
+		func(buf *bytes.Buffer) (*Tap, error) {
+			parsedAddr := &Tap{
+				ChainParams: &TestNet3Tap,
+			}
+			return parsedAddr, parsedAddr.Decode(buf)
+		},
+		func(parsedAddr *Tap, unknownTypes tlv.TypeMap) {
+			require.Equal(
+				t, unknownTypes, parsedAddr.unknownOddTypes,
+			)
+
+			// The address should've changed, to make sure the
+			// unknown value was taken into account when creating
+			// the serialized address.
+			parsedAddrString, err := parsedAddr.EncodeAddress()
+			require.NoError(t, err)
+
+			require.NotEqual(t, knownAddrString, parsedAddrString)
+
+			parsedAddr.unknownOddTypes = nil
+			parsedAddr.assetGen = asset.Genesis{}
```

Comment:
Same Q, why does the genesis need to be cleared for the comparison? Explanatory comment would be useful if it is needed.